aws_topology/stackstate_checks/aws_topology/resources/kinesis.py

```diff
@@ -1,88 +1,118 @@
-from .utils import make_valid_data, create_arn as arn
+from collections import namedtuple
+from .utils import (
+    make_valid_data,
+    create_arn as arn,
+    client_array_operation,
+    set_required_access_v2,
+    transformation,
+)
+from schematics import Model
+from schematics.types import StringType, ModelType
 from .registry import RegisteredResourceCollector
 
 
 def create_arn(region=None, account_id=None, resource_id=None, **kwargs):
-    return arn(resource='kinesis', region=region, account_id=account_id, resource_id='stream/' + resource_id)
+    return arn(resource="kinesis", region=region, account_id=account_id, resource_id="stream/" + resource_id)
+
+
+StreamData = namedtuple("StreamData", ["stream", "tags"])
+
+
+class Stream(Model):
+    StreamName = StringType()
+    StreamARN = StringType()
+
+
+class StreamDescriptionSummary(Model):
+    StreamDescriptionSummary = ModelType(Stream)
 
 
 class KinesisCollector(RegisteredResourceCollector):
     API = "kinesis"
     API_TYPE = "regional"
     COMPONENT_TYPE = "aws.kinesis"
-    CLOUDFORMATION_TYPE = 'AWS::Kinesis::Stream'
+    CLOUDFORMATION_TYPE = "AWS::Kinesis::Stream"
+
+    @set_required_access_v2("kinesis:ListTagsForStream")
+    def collect_tags(self, stream_name):
+        self.client.list_tags_for_stream(StreamName=stream_name).get("Tags", [])
+
+    @set_required_access_v2("kinesis:DescribeStreamSummary")
+    def collect_stream_description(self, stream_name):
+        return self.client.describe_stream_summary(StreamName=stream_name)
+
+    def collect_stream(self, stream_name):
+        data = self.collect_stream_description(stream_name)
+        tags = self.collect_tags(stream_name)
```

Comment:
add or [] for tags
add or { 'StreamName': stream_name, 'StreamARN': self.agent.create_arn('AWS::Bla::Bla', self.location_info, stream_name) }